x-pack/plugins/maps/public/classes/layers/layer.tsx

```diff
@@ -90,6 +90,7 @@ export interface ILayer {
   supportsLabelsOnTop: () => boolean;
   showJoinEditor(): boolean;
   getJoinsDisabledReason(): string | null;
+  isFittable(): Promise<boolean>;
```

Comment:
How about naming `isFittable` to `supportsFitToBounds` so its more similar to the existing method on `ISource`?

Reply:
as discussed. changed implementation to push visibility-check into `isFittable` and delegate to source to check if bounds-retrieval is supported.